src/storage/archive_index.rs

```diff
@@ -2,13 +2,15 @@ use crate::error::Result;
 use crate::storage::{compression::CompressionAlgorithm, FileRange};
 use anyhow::{bail, Context as _};
 use memmap2::MmapOptions;
+use rusqlite::{Connection, OptionalExtension};
 use serde::de::DeserializeSeed;
 use serde::de::{IgnoredAny, MapAccess, Visitor};
 use serde::{Deserialize, Deserializer, Serialize};
-use std::collections::HashMap;
-use std::fmt;
-use std::path::Path;
-use std::{fs, io};
+use std::{collections::HashMap, fmt, fs, fs::File, io, io::Read, path::Path};
+
+use super::sqlite_pool::SqliteConnectionPool;
+
+static SQLITE_FILE_HEADER: &[u8] = b"SQLite format 3\0";
```

Comment:
This could be kept inside the `is_sqlite_file` function.

Reply:
it's also used in the tests, how would you handle it for these?